homeassistant/components/rainmachine.py

```diff
@@ -0,0 +1,113 @@
+"""
+This component provides support for RainMachine sprinkler controllers.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/rainmachine/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+from requests.exceptions import ConnectTimeout
+
+from homeassistant.helpers import config_validation as cv
+from homeassistant.const import (
+    CONF_EMAIL, CONF_IP_ADDRESS, CONF_PASSWORD, CONF_PORT, CONF_SSL)
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['regenmaschine==0.4.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_RAINMACHINE = 'data_rainmachine'
+DOMAIN = 'rainmachine'
+
+NOTIFICATION_ID = 'rainmachine_notification'
+NOTIFICATION_TITLE = 'RainMachine Component Setup'
+
+CONF_ATTRIBUTION = 'Data provided by Green Electronics LLC'
+
+DEFAULT_PORT = 8080
+DEFAULT_SSL = True
+
+MIN_SCAN_TIME_LOCAL = timedelta(seconds=1)
+MIN_SCAN_TIME_REMOTE = timedelta(seconds=1)
+MIN_SCAN_TIME_FORCED = timedelta(milliseconds=100)
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema(
+        vol.All(
+            cv.has_at_least_one_key(CONF_IP_ADDRESS, CONF_EMAIL),
+            {
+                vol.Exclusive(CONF_IP_ADDRESS, 'auth'): cv.string,
+                vol.Exclusive(CONF_EMAIL, 'auth'):
+                    vol.Email(),  # pylint: disable=no-value-for-parameter
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+            }))
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up an Arlo component."""
+    from regenmaschine import Authenticator, Client
+    from regenmaschine.exceptions import HTTPError
+
+    _LOGGER.debug('Config data: %s', config)
+
+    conf = config[DOMAIN]
+    ip_address = conf.get(CONF_IP_ADDRESS, None)
```

Comment:
`None` is the default value returned.

Reply:
Not sure what you mean by this; all other components I've looked at return `True` or `False`.

Reply:
`dict.get(key)` by default returns `None` if the key is missing in the dict. So please just write `conf.get(CONF_IP_ADDRESS)`.

Sorry about not being clear.

Reply:
Got it! Will update.